packages/flutter/lib/src/widgets/animated_cross_fade.dart

```diff
@@ -354,9 +358,12 @@ class _AnimatedCrossFadeState extends State<AnimatedCrossFade> with TickerProvid
       enabled: true, // Top widget always has its animations enabled.
       child: ExcludeSemantics(
         excluding: false, // Always publish semantics for the widget that's fading in.
-        child: FadeTransition(
-          opacity: topAnimation,
-          child: topChild,
+        child: ExcludeFocus(
```

Comment:
This is still not needed: the `ExcludeFocus` is a no-op when `excluding` is false.

Reply:
I get an error in testing when I remove ExcludeFocus, but I don't know why. Can you please help me?
<details>
<summary>error</summary>

00:04 +5 -1: AnimatedCrossFade preserves widget state [E]                                          
  Test failed. See exception logs above.
  The test description was: AnimatedCrossFade preserves widget state
  
00:04 +5 -1: AnimatedCrossFade switches off TickerMode and semantics on faded out widget           
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure object was thrown running a test:
  Expected: <true>
  Actual: <false>

When the exception was thrown, this was the stack:
#4      main.<anonymous closure> (file:///Users/zuvola/Documents/Projects/etc/flutter/packages/flutter/test/widgets/animated_cross_fade_test.dart:317:5)

</details>

Reply:
Ahh, I see. I'm wrong: you actually do need the `ExcludeFocus` with `excluding` set to false here, because otherwise the change in the widget hierarchy causes the children's state to be rebuilt, which is undesirable. It's true that it is a no-op when it comes to focus state, but not with respect to the widget tree topology.

Reply:
OK, I understand. Thank you for the detailed explanation.